yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -229,7 +229,11 @@ export class MemoryArchiverStore implements ArchiverDataStore {
    * @returns True if the operation is successful (always in this implementation).
    */
   public addL2ContractPublicData(data: ContractPublicData[], blockNum: number): Promise<boolean> {
-    this.contractPublicData[blockNum] = data;
+    if (this.contractPublicData[blockNum]?.length) {
+      this.contractPublicData[blockNum]?.push(...data);
+    } else {
+      this.contractPublicData[blockNum] = data;
```

Comment:
```suggestion
      this.contractPublicData[blockNum] = [...data];
```
Just so that we don't modify the first `ContractPublicData` for a block number when we insert the 2nd one.